apps/antalmanac/src/components/Map/Map.tsx

```diff
@@ -176,13 +180,15 @@ export default function CourseMap() {
                 {/** Menu floats above the map. */}
                 <Paper sx={{ zIndex: 400, position: 'relative', my: 2, mx: 6.942, marginX: '15%', marginY: 8 }}>
                     <Tabs value={selectedDayIndex} onChange={handleChange} variant="fullWidth" sx={{ minHeight: 0 }}>
-                        {days.map((day) => (
-                            <Tab
-                                key={day}
-                                label={day || 'All'}
-                                sx={{ padding: 1, minHeight: 'auto', minWidth: '10%', p: 1 }}
-                            />
-                        ))}
+                        {days
+                            .filter((day) => hasWeekendCourse || (day !== 'Sat' && day !== 'Sun'))
```

Comment:
Instead of doing the filter on `days` here can you move the variable to be initialized after `hasWeekendCourse`? Looking at this rn, it seems like `days` is scoped globally in the file for no reason, and IMO this nested filter+map is slightly harder to read than just a `map`. 

Also while you're at it, can you change `day || 'All'` to just `day` and replace the empty string in the initial list with `'All'`?

Reply:
Days is correctly scoped globally. If it's defined inside the component, then it's reconstructed on every single render, i.e. whenever the store or state is mutated.

Reply:
Oh, and "" instead of "All" is a deliberate choice because the filtering of days does a match, e.g.

```ts
event.day.includes(day)
```
"" is chosen to represent "All days" because all strings will contain an empty string.

Reply:
Good point since I think it is used like that in other places in the codebase, but for this instance I'm pretty sure it can be "All" because the `Tabs` selector only passes on the index of the choice, and having the string there accurately represent the label is better IMO as it's slightly more readable. I also think the `event.day.includes(day)` matching empty string is kind of a hack and should just be like `day=='all' || event.day.includes(day)` for clarity but changing that everywhere in the codebase is out of scope for this PR

Reply:
@KevinWu098 I think this is reasonable

Reply:
Also I kinda disagree about days being scoped globally, because I don't think the performance difference really matters here since it's just initializing a list with strings, but the current solution in this PR is a bit better than it was before. One additional change I think we should make is making the variables all caps as is the convention for hard-coded constants. I guess that's one disadvantage of `const` in JS because a const list can get modified but in this case we want to make it clear the list won't be modified.